samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor

```diff
@@ -175,93 +128,157 @@
     public bool IsOpened { get; set; } = false;
-    public ListNode nodeData { get; set; }
-
-    private enum ActionType { Nothing, Read, Write, Call, Publish };
-    private ActionType TypeOfAction { get; set; } = ActionType.Nothing;
-    private string Response { get; set; } = string.Empty;
-    private string Value { get; set; } = string.Empty;
+    public ListNode NodeData { get; set; }
+    
 
-
-    public void PagerPageChanged(int page)
+    /// <summary>
+    /// PagerPageChanged
+    /// </summary>
+    /// <param name="page"></param>
+    /// <returns></returns>
+    public async void PagerPageChanged(int page)
     {
-        pagedNodeList = NodeList.GetPaged(page, 10);
+        PagedNodeList = NodeList.GetPaged(page, 10);
+        foreach (var node in PagedNodeList.Results)
+        {
+            // fetch the actual value
+            if (node.NodeClass == NodeClass.Variable)
+            {
+                node.Value = await BrowseManager.ReadValueAsync(endpointId, node.Id);
+            }
+        }
         NavigationManager.NavigateTo("/browser/" + supervisorId + "/" + endpointId + "/" + page);
     }
 
+    /// <summary>
+    /// OnInitializedAsync
+    /// </summary>
+    /// <returns></returns>
     protected override async Task OnInitializedAsync()
     {
         await BrowseTreeAsync(string.Empty, new List<string>(), BrowseDirection.Forward);
     }
 
+    /// <summary>
+    /// GetTreeAsync
+    /// </summary>
+    /// <param name="id"></param>
+    /// <param name="parentId"></param>
+    /// <returns></returns>
     private async Task GetTreeAsync(string id, List<string> parentId)
     {
         await BrowseTreeAsync(id, parentId, BrowseDirection.Forward);
     }
 
+    /// <summary>
+    /// GetTreeBackAsync
+    /// </summary>
+    /// <param name="id"></param>
+    /// <param name="parentId"></param>
+    /// <returns></returns>
     private async Task GetTreeBackAsync(string id, List<string> parentId)
     {
         await BrowseTreeAsync(id, parentId, BrowseDirection.Backward);
         NavigationManager.NavigateTo("/browser/" + supervisorId + "/" + endpointId);
     }
 
+    /// <summary>
+    /// BrowseTreeAsync
+    /// </summary>
+    /// <param name="id"></param>
+    /// <param name="parentId"></param>
+    /// <param name="direction"></param>
+    /// <returns></returns>
     private async Task BrowseTreeAsync(String id, List<string> parentId, BrowseDirection direction)
     {
         NodeList = await BrowseManager.GetTreeAsync(endpointId, id, parentId, supervisorId, direction);
-        Page = "1";
-        pagedNodeList = NodeList.GetPaged(Int32.Parse(Page), 10);
-    }
+        var publishedNodes = await Publisher.PublishedAsync(endpointId);
 
-    private void OpenDrawer(ListNode node)
-    {
-        IsOpened = true;
-        nodeData = node;
-    }
+        foreach (var node in NodeList.Results)
+        {
+            // fetch the actual value
```

Comment:
>// fetch the actual value [](start = 12, length = 25)

this comment does not describe the loop action here?

Reply:
fixed